src/v/cloud_storage/topic_manifest_downloader.cc

```diff
@@ -184,15 +271,24 @@ topic_manifest_downloader::find_manifests(
           remote, bucket, parent_retry, deadline, backoff, root));
     }
     auto prefixed_res = co_await ss::when_all_succeed(futs.begin(), futs.end());
+    bool has_prefix_list_error = false;
     for (const auto& r : prefixed_res) {
         if (r.has_error()) {
-            co_return error_outcome::manifest_download_error;
+            vlog(
+              cst_log.error,
+              "Finding topic manifest resulted in listing error for prefixed "
+              "roots: {}",
+              r.error());
+            has_prefix_list_error = true;
         }
         for (const auto& item : r.value()) {
```

Comment:
when `r.has_error()` is true this will fall through and `r.value()` will result in undefined behavior.

Reply:
Yikes! done